recipe/meta.yaml

```diff
@@ -6,63 +6,65 @@ package:
   version: {{ version }}
 
 source:
-  fn: opencv-{{ version }}.tar.gz
-  url: https://github.com/opencv/opencv/archive/{{ version }}.tar.gz
-  sha256: f1b87684d75496a1054405ae3ee0b6573acaf3dad39eaf4f1d66fdd7e03dc852
+  - url: https://github.com/opencv/opencv/archive/{{ version }}.tar.gz
+    fn: opencv-{{ version }}.tar.gz
+    sha256: f1b87684d75496a1054405ae3ee0b6573acaf3dad39eaf4f1d66fdd7e03dc852
+    patches:  # [win]
+      - 01_win_fix_cmake_config_sufix.patch  # [win]
+  - url: https://github.com/opencv/opencv_contrib/archive/{{ version }}.tar.gz
+    fn: opencv_contrib-{{ version }}.tar.gz
+    sha256: 298c69ee006d7675e1ff9d371ba8b0d9e7e88374bb7ba0f9d0789851d352ec6e
+    folder: opencv_contrib-{{ version }}
 
 build:
-  number: 200
+  number: 204
   # Python2.7 support dropped: https://github.com/opencv/opencv/issues/8481
   skip: true  # [win and py27]
-  features:                    # [not win]
+  features:  # [not win]
     - blas_{{ blas_variant }}  # [not win]
 
 requirements:
   build:
     - python
     - toolchain
-    - pkgconfig                    # [unix]
-    # For sha256 comparisons of opencv_contrib
-    - openssl 1.0.*                # [unix]
-    # For downloading opencv_contrib
-    - curl >=7.44.0,<8
     # For applying patches
-    - git                          # [win]
+    - m2-patch  # [win]
+    - pkgconfig  # [unix]
     - cmake
-    - numpy 1.8.*   # [not (win and (py35 or py36))]
-    - numpy 1.9.*   # [win and py35]
+    - numpy 1.8.*  # [not (win and (py35 or py36))]
+    - numpy 1.9.*  # [win and py35]
     - numpy 1.11.*  # [win and py36]
-    - hdf5 1.10.1         # [unix]
+    - hdf5 1.10.1  # [unix]
     - eigen 3.2.*
-    - jasper                       # [unix]
+    - jasper  # [unix]
     - zlib 1.2.11
     - jpeg 9*
     - libtiff >=4.0.8,<4.0.10
     - libwebp 0.5.*
-    - harfbuzz 1.7.*               # [unix]
+    - harfbuzz 1.7.*  # [unix]
     - libpng >=1.6.32,<1.6.35
-    - openblas 0.2.20|0.2.20.*     # [not win]
+    - openblas 0.2.20|0.2.20.*  # [not win]
     - blas 1.1 {{ blas_variant }}  # [not win]
-    - ffmpeg >=3.2.3,<3.2.6        # [not win]
-    - qt 5.6.*                     # [not osx]
+    - ffmpeg >=3.2.3,<3.2.6  # [not win]
+    - qt 5.6.*  # [not osx]
```

Comment:
We should look at adopting the new pinning strategy.

ref: https://conda-forge.org/docs/meta.html#pinning-packages

Reply:
Ok, I'm new to conda build/package system (I'm becoming found of it) but I'll need some help.
I've looked at this, but I'm missing some understanding of the pinning mechanism. what is it that you'd expect here?
1- what's the role of https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml?
1.1 - it aims to provide a conda-forge compatible pinning snapshot for conda-forge packages ?
1.1.1 - if so how does it integrate with libraries build from other channels ?
1.1 are the files .ci_support/linux_python2.7.yaml  added by conda-smithy relate to conda_build_config.yaml ?

Reply:
Also, how will I pinn it ? just remove the versions, as they will be filled in by conda_build_config.yaml ?